wgsl/index.bs

```diff
@@ -6807,12 +6803,13 @@ The type of the return value [=shader-creation error|must=] match the return typ
 
 ### Discard Statement ### {#discard-statement}
 
-A <dfn dfn-for="statement">discard</dfn> statement immediately ends execution of a fragment shader invocation and throws away the fragment.
+A <dfn dfn-for="statement">discard</dfn> statement converts the invocation into
+a [=helper invocation=] and throws away the fragment.
 The `discard` statement [=shader-creation error|must=] only be used in a [=fragment=] shader stage.
 
 More precisely, executing a `discard` statement will:
 
-* immediately terminate the current invocation, and
+* convert the current invocation into a [=helper invocation=], and
 * prevent evaluation and generation of a [=return value=] for the [=entry point=], and
```

Comment:
I think delete the clause about evaluation of the return?  I think the evaluation of the return should occur, e.g. might have a function call in it.  But it should be subject to the constraints of a helper invocation.